torchx/mix.exs

```diff
@@ -3,6 +3,7 @@ defmodule Torchx.MixProject do
 
   @source_url "https://github.com/elixir-nx/nx"
   @version "0.1.0-dev"
+  @default_libtorch_version File.read!(".default_libtorch_version")
```

Comment:
Why a file? I don't think Elixir makes it easy to configure a file inside a dependency. Instead it uses environment variables: `{:torchx, ..., system_env: %{"LIBTORCH_TARGET" => "cpu"}}`.

Reply:
So we can have the same source of truth for the makefile as well. This isn't supposed to be configurable anyway, it's just for better defaults on Torchx's side

Reply:
I added this because I ran into the exact issue which this prevents, in which mix.exs is updated but the makefile default value is not

Reply:
Oh, it is a file because you want to share it with the makefile. This is not a problem! It is possible to pass environment variables to the makefile. See https://hexdocs.pm/elixir_make. We should pass the LIBTORCH_DIR downstream and assume it is set in the `Makefile`.

Reply:
Something like this:

```elixir
@libtorch_version "1.9.1"
@libtorch_target "cpu"
@libtorch_cache Path.expand(__DIR__, "cache/libtorch-#{@libtorch_version}-#{@libtorch_target}")

def project do
  [
    ...,
    make_env: %{"LIBTORCH_DIR" => System.get_env("LIBTORCH_DIR") || @libtorch_cache}
  ]
```

Reply:
Then cache_dir is `Path.dirname(@libtorch_cache)`. :)

Reply:
Oh, nice! I'll change this as soon as I get back on the PC. Besides this, is eveything ok with this PR? I want to kerged it asap because it blocks further development a little bit

Reply:
Yes, everything else looks good. If you want to wait for this, then just revert this change. The important is that we change the filename BEFORE we bump the version/target again.

Reply:
No worries, we can merge this when it's actually ready :)